### PR TITLE
Handle Matrix 500 erros in `matrixMessageSend[Global]Epic`

### PR DESCRIPTION
**Pull request by palango**
**Thank you for submitting this pull request :)**

Part of #2314

**Short description**

Handle and retry matrix 500 errors and retry them.

**Definition of Done**

- [ ] Steps to manually test the change have been documented
- [ ] Acceptance criteria are met
- [ ] Change has been manually tested by the reviewer (dApp) 

**Steps to manually test the change (dApp)**

1.
2.

### CODE REVIEW

File: raiden-ts/src/transport/epics/messages.ts [hunk 1]
```diff
@@ -86,6 +89,10 @@ export function matrixMessageSendEpic(
         { log, latest$, config$ },
         true, // allowRtc
       ).pipe(
+        retryWaitWhile(
```
Comment (andrevmatos) [inline raiden-ts/src/transport/epics/messages.ts:92]: `waitMemberAndSend$` already has a retry mechanism embedded. I think it's better to extend it than to add another retry layer here.

File: raiden-ts/src/transport/epics/messages.ts [hunk 2]
```diff
@@ -140,6 +147,10 @@ export function matrixMessageGlobalSendEpic(
             '',
           );
         }),
+        retryWaitWhile(
+          exponentialBackoff(config.pollingInterval, config.httpTimeout),
+          (err, count) => err?.httpStatus !== 429 || err?.httpStatus !== 500 || count > 3,
```
Comment (andrevmatos) [inline raiden-ts/src/transport/epics/messages.ts:152]:
```suggestion
          (err, count) => ![429, 500].includes(err?.httpStatus) || count > 3,
```
Or `!==` + `||` on 2 different constants is always true

**Review by andrevmatos — COMMENTED** [review]
just some small issues, almost gtg, tyvm

File: raiden-ts/src/transport/epics/messages.ts (hunk 1 above)
Comment (palango) [reply raiden-ts/src/transport/epics/messages.ts]: Good point! Fixed

**Review by andrevmatos — APPROVED** [review]
lgtm, ty

**Pull request merged**